### PR TITLE
TF 2.5.* support

### PR DESCRIPTION
**Pull request by negvet**
### Changes

TF2.5.* supported

### Related tickets

62338

### Tests

Graph tests are updated. Now graphs for 2.4.* and for 2.5.* versions are stored in separate folders.

### CODE REVIEW

File: nncf/version.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
 __version__ = '2.1.0'
 BKC_TORCH_VERSION = '1.9.1'
 BKC_TORCHVISION_VERSION = '0.10.1'
-BKC_TF_VERSION = '2.4.*'
+BKC_TF_VERSION = '2.5.0'
```
Comment (alexsu52) [inline nncf/version.py:4]: We must support 2.5.*

**Review by alexsu52 — CHANGES_REQUESTED** [review]

File: nncf/version.py (hunk 1 above)
Comment (negvet) [reply nncf/version.py]: Updated

File: README.md [hunk 2]
```diff
@@ -150,9 +150,9 @@ See [third_party_integration](./third_party_integration) for examples of code mo
 - Python\* 3.6.2 or later
 - Supported frameworks:
   - PyTorch\* >=1.5.0, <=1.9.1 (1.8.0 not supported)
-  - TensorFlow\* 2.4.3
+  - TensorFlow\* >=2.4.3, <=2.5.2
 
-This repository is tested on Python* 3.6.2+, PyTorch* 1.9.1 (NVidia CUDA\* Toolkit 10.2) and TensorFlow* 2.4.3 (NVidia CUDA\* Toolkit 11.0).
+This repository is tested on Python* 3.6.2+, PyTorch* 1.9.1 (NVidia CUDA\* Toolkit 10.2) and TensorFlow* 2.5.2 (NVidia CUDA\* Toolkit 11.2).
```
Comment (alexsu52) [inline README.md:155]:
```suggestion
This repository is tested on Python* 3.6.2+, PyTorch* 1.9.1 (NVidia CUDA\* Toolkit 10.2) and TensorFlow* 2.5.3 (NVidia CUDA\* Toolkit 11.2).
```

File: examples/tensorflow/object_detection/main.py [hunk 3]
```diff
@@ -378,6 +379,11 @@ def validate_fn(model, **kwargs):
         compression_ctrl.export_model(save_path, save_format)
         logger.info("Saved to {}".format(save_path))
 
+    # Due to https://github.com/tensorflow/tensorflow/issues/50487
+    # pylint: disable=protected-access
+    # pylint: disable=too-many-statements
+    atexit.register(strategy._extended._collective_ops._pool.close)
```
Comment (alexsu52) [inline examples/tensorflow/object_detection/main.py:385]: Is it fixed in the TF 2.5.3?

File: nncf/tensorflow/__init__.py [hunk 4]
```diff
@@ -19,11 +19,17 @@
...
 if not tensorflow_version.startswith(BKC_TF_VERSION[:-2]):
-    raise RuntimeError(
-         'NNCF only supports tensorflow=={bkc}, while current tensorflow version is {curr}'.format(
+    import warnings
+    warnings.warn("NNCF provides best results with tensorflow=={bkc}, "
+                   "while current tensorflow version is {curr} - consider switching to tensorflow=={bkc}".format(
          bkc=BKC_TF_VERSION,
          curr=tensorflow.__version__
     ))
+elif not ('2.4.3' <= tensorflow_version <= '2.5.2'):
```
Comment (alexsu52) [inline nncf/tensorflow/__init__.py:28]: I guess we will support `tensorflow >=2.4.0, ==2.5.*`

**Review by alexsu52 — CHANGES_REQUESTED** [review]
Please provide results of the pre-commit tests for TF 2.4.* and TF 2.5.*

File: examples/tensorflow/object_detection/main.py (hunk 3 above)
Comment (negvet) [reply examples/tensorflow/object_detection/main.py:385]: No, it is not

File: README.md [hunk 5]
```diff
@@ -150,9 +150,9 @@ See [third_party_integration](./third_party_integration) for examples of code mo
 - Python\* 3.6.2 or later
 - Supported frameworks:
   - PyTorch\* >=1.5.0, <=1.9.1 (1.8.0 not supported)
-  - TensorFlow\* 2.4.3
+  - TensorFlow\* >=2.4.3, <=2.5.2
```
[the remark this answers is not in the document]
Comment (negvet) [reply README.md]: Fixed

File: README.md (hunk 2 above)
Comment (negvet) [reply README.md]: Fixed

File: nncf/tensorflow/__init__.py (hunk 4 above)
Comment (negvet) [reply nncf/tensorflow/__init__.py]: Sure

**Comment by negvet** [comment]
Jenkins please retry a build

File: examples/tensorflow/object_detection/main.py [hunk 6]
```diff
@@ -266,6 +267,7 @@ def evaluate(test_step, metric, test_dist_dataset, num_batches, print_freq):
     return result
 
 
+# pylint: disable=too-many-statements
```
Comment (alexsu52) [inline examples/tensorflow/object_detection/main.py:270]: Please, refactor the `run` function to resolve the pylint issue.

**Review by alexsu52 — CHANGES_REQUESTED** [review]

File: examples/tensorflow/object_detection/main.py (hunk 6 above)
Comment (negvet) [reply examples/tensorflow/object_detection/main.py]: Thanks for the comment, refactored.

**Review by alexsu52 — APPROVED** [review]

**Comment by negvet** [comment]
Jenkins please retry a build

**Review by vshampor — APPROVED** [review]

**Pull request merged**